src/pybind/mgr/balancer/module.py

```diff
@@ -324,6 +324,7 @@ class Module(MgrModule):
     last_optimize_started = ''
     last_optimize_duration = ''
     optimize_result = ''
+    no_optimization_needed = False
```

Comment:
Just out of curiosity: why do you use`no_optimization_needed`? I'm wondering if `optimization_needed` would be easier. But there might be a particular reason I don't see.

Reply:
I used `no_optimization_needed` flag becuase I want this flag to be true only when we get this result - `Unable to find further optimization,'or pool(s) pg_num is decreasing,'or distribution is already perfect` from the balancer. So I just need to add this flag here only, otherwise If I had to use `optimize_needed` flag, then I need to add that flag at multiple places in the code wherever the status is different than the above I mentioned.

Reply:
Thank you for the explanation :+1: :slightly_smiling_face: